jlm/hls/opt/cne.cpp

```diff
@@ -573,10 +575,10 @@ divert_lambda(rvsdg::StructuralNode * node, cnectx & ctx)
 static void
 divert_phi(rvsdg::StructuralNode * node, cnectx & ctx)
 {
-  JLM_ASSERT(is<llvm::phi::operation>(node));
+  auto & phi = dynamic_cast<PhiNode &>(*node);
```

Comment:
`util::AssertedCast<>`

Reply:
changed (TBH, why not let it throw std::bad_cast?)